### PR TITLE
Fix use after free in VHPI

### PR DESCRIPTION
**Pull request by ktbarrett**
Fixing test for #3307. New PR since I can't push to that branch.

### CODE REVIEW

File: tests/test_cases/test_iteration_vhdl/test_iteration.py [hunk 1]
```diff
@@ -120,10 +116,7 @@ async def iteration_loop():
 
 # GHDL unable to access record types (gh-2591)
 @cocotb.test(
-    expect_fail=(
-        cocotb.SIM_NAME.lower().startswith("riviera")
-        or cocotb.SIM_NAME.lower().startswith("aldec")
-    ),
+    expect_fail=cocotb.SIM_NAME.lower().startswith("aldec"),
```
Comment (imphil) [inline tests/test_cases/test_iteration_vhdl/test_iteration.py:119]: Remove this line as well, aldec is a deprecated alias for riviera.

**Comment by imphil** [comment]
Btw, it looks like we solved a long-standing mystery by accident here. Nice!

File: tests/test_cases/test_iteration_vhdl/test_iteration.py [hunk 2]
```diff
@@ -45,15 +45,11 @@ def total_object_count():
...
     ):
         return 34569
 
-    # Riviera-PRO
-    if SIM_NAME.startswith("riviera"):
-        # The expected result from Riviera-PRO 2019.10 on.
-        return 27359
-
     # Active-HDL
     if SIM_NAME.startswith("aldec"):
```
[the remark this answers is not in the document]
Comment (marlonjames) [reply tests/test_cases/test_iteration_vhdl/test_iteration.py:54]: The problem is that the simulator name given through VPI/VHPI is something like "aldec activehdl".
If we use `startswith()` then we need "aldec" as part of it.

[the remark this answers is not in the document]
Comment (imphil) [reply tests/test_cases/test_iteration_vhdl/test_iteration.py:54]: Ah true, we're not matching against SIM -- I got confused. Clearly time for some cleanup in that area, but that's not for this PR. Ignore the noise, @ktbarrett!

**Comment by ktbarrett** [comment]
@Forty-Bot No worries, this PR should have all the bits to get it merged. You're still marked as author on the original commit.

**Review by imphil — APPROVED** [review]
Great, thanks @Forty-Bot and @ktbarrett!

**Review by marlonjames — APPROVED** [review]

**Pull request merged**